### PR TITLE
Refactoring and fix JSDocs

### PR DESCRIPTION
**Pull request by MatteoMolinari93**
First refactoring round before adding support for credentials objects

### CODE REVIEW

File: gmail-tester.js [hunk 1]
```diff
@@ -2,6 +2,104 @@ const gmail = require("./gmail");
 const fs = require("fs");
 const { google } = require("googleapis");
 
+module.exports = {
```
Comment (levz0r) [inline gmail-tester.js:5]: General note on changes in this file: In order to make this PR more readable, I think it is better to preserve the original order of the functions' declaration. What do you think?

File: gmail-tester.js [hunk 2]
```diff
@@ -2,6 +2,104 @@ const gmail = require("./gmail");
...
+ * @param {boolean} options.include_body - Return message body string.
+ * @param {string} options.from - Filter on the email address of the receiver.
+ * @param {string} options.to - Filter on the email address of the sender.
+ * @param {string} options.subject - Filter on the subject of the email.
+ * @param {Object} options.before - Date. Filter messages received _after_ the specified date.
+ * @param {Object} options.after - Date. Filter messages received _before_ the specified date.
+ */
+async function get_messages(credentials_json, token_path, options) {
+  try {
+    return await _get_recent_email(credentials_json, token_path, options);
```
Comment (levz0r) [inline gmail-tester.js:64]: For debugging purposes, I think it is better to save the results of `_get_recent_email` into a variable and return it. Would like to know your opinion on this.

File: gmail.js [hunk 3]
```diff
@@ -24,10 +20,7 @@ async function authorize(credentials, token_path) {
   );
   // Check if we have previously stored a token.
   try {
-    const token = fs.readFileSync(
-      token_path || path.resolve(__dirname, TOKEN_PATH)
-    );
-    oAuth2Client.setCredentials(JSON.parse(token));
+    oAuth2Client.setCredentials(tokenStore.get(token_path));
```
Comment (levz0r) [inline gmail.js:23]: Similarly to my previous comment, better save the result of `tokenStore.get` into a variable and then pass it to `setCredentials`.

File: token-store.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+ * Stores the token in the specified path or in default path.
+ * @param {Object} token Token
+ * @param {String | null} token_path Path
+ */
+function store(token, token_path) {
+  fs.writeFileSync(
+    token_path || path.resolve(__dirname, TOKEN_PATH),
+    JSON.stringify(token)
+  );
+}
```
Comment (levz0r) [inline token-store.js:39]: Please preserve EOL.

File: token-store.js [hunk 5]
```diff
@@ -0,0 +1,39 @@
+const fs = require('fs');
+const path = require('path');
+
+// The file token.json stores the user's access and refresh tokens, and is
+// created automatically when the authorization flow completes for the first
+// time.
+const TOKEN_PATH = "token.json";
+
+module.exports = {
```
Comment (levz0r) [inline token-store.js:9]: Exports usually go to the end of the file.

**Review by levz0r — COMMENTED** [review]
Thank you for the PR. Please see my comments.

File: gmail-tester.js (hunk 1 above)
Comment (MatteoMolinari93) [reply gmail-tester.js:5]: Yes I agree. As a developer I felt and think that the exports and the exported functions are better at the top, so I changed it. But at the end is your call as the owner to define a general rule for this repo, so please think about it.

File: token-store.js (hunk 5 above)
Comment (MatteoMolinari93) [reply token-store.js]: Moved to the bottom for consistency

File: gmail-tester.js (hunk 2 above)
Comment (MatteoMolinari93) [reply gmail-tester.js]: I disagree. I think our goal should be to keep the code as readable as possible and not "debug ready". Additions meant only to make the debug easier complicate the understanding of the code base for new users and discourage them to make changes. (Ok in this case I'm a bit exagerating, it's just an inline variable, but as rule of thumb :D)

File: gmail.js (hunk 3 above)
Comment (MatteoMolinari93) [reply gmail.js:23]: Same response

**Review by levz0r — APPROVED** [review]

**Pull request merged**